Library/ViewModels/RewardsCollectionViewModel.swift

```diff
@@ -215,6 +276,19 @@ private func shouldNavigateToReward(project: Project, reward: Reward, refTag _:
   project.state == .live && (!userIsBacking(reward: reward, inProject: project) || reward.hasAddOns)
 }
 
+private func shouldTriggerEditRewardPrompt(_ data: PledgeViewData) -> Bool {
+  // If the user is not backing the project then there is no need to show the prompt.
+  guard
+    userIsBackingProject(data.project),
+    let backing = data.project.personalization.backing
+  else { return false }
+
+  let rewardUnchanged = data.rewards.first?.id == backing.reward?.id
```

Comment:
Will the first reward id always be equal to the backing.reward.id?

Reply:
Yeah for now I know that it's the first one but this could probably be improved to be more explicit about what the base reward is in the `PledgeViewData` type. Let's aim to improve this once we get this release out 👍